### PR TITLE
chore: Prevent PR merge with `do not merge` labels 

### PR DESCRIPTION
**Pull request by kdheepak**
This PR prevents merges if a `do not merge` label is added on GitHub.

This PR also fixes an issue that allows checking PR titles with backticks in the title string.

### CODE REVIEW

**Comment by kdheepak** [comment]
With the current title of this PR, the check-breaking-change-label step will fail; but this PR will fix it and going forward this won't be an issue. So this PR can be merged even if that step fails.

**Review by joshka — APPROVED** [review]

File: .github/workflows/check-pr.yml
Comment (orhun) [inline .github/workflows/check-pr.yml:55]: We could probably do something like this here:

```
if [[ "${{ github.event.pull_request.title }}" =~ $pattern ]]; then
```

But I'm too late for the review, can you guys wait a bit before merge?

Comment (kdheepak) [reply .github/workflows/check-pr.yml:55]: Sorry about that! Next time I'll wait longer for merges.

**Pull request merged**
